internal/client/client.go

```diff
@@ -57,23 +48,23 @@ func (c *client) Put(ctx context.Context, key, val string, opts ...clientv3.OpOp
 	return genericPP[string, string, clientv3.OpOption, clientv3.PutResponse](ctx, c.log, c, c.kv.Put, key, val, opts...)
 }
 
-func (c *client) PutIfNotExists(ctx context.Context, key, val string, opts ...clientv3.OpOption) (*clientv3.PutResponse, error) {
-	fn := func(ctx context.Context, key, val string, opts ...clientv3.OpOption) (*clientv3.PutResponse, error) {
+func (c *client) PutIfNotExists(ctx context.Context, key, val string, opts ...clientv3.OpOption) (bool, error) {
+	var put bool
+	err := generic(ctx, c.log, c, func(ctx context.Context) error {
 		tif := clientv3.Compare(clientv3.CreateRevision(key), "=", 0)
 		tthen := clientv3.OpPut(key, val, opts...)
-		gresp, err := c.kv.Txn(ctx).If(tif).Then(tthen).Commit()
+		gresp, err := c.kv.Txn(ctx).If(tif).Then(tthen).
+			Commit()
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		if gresp == nil || !gresp.Succeeded {
-			return nil, clienterrors.NewKeyAlreadyExists(key)
-		}
+		put = gresp != nil && gresp.Succeeded
```

Comment:
why did you decide to remove the NewKeyAlreadyExists err?

Reply:
Because we are changing to returning a `bool` instead it is successful. This simplifies things for out internal handling, without needs to operator over error types.